src/courses/views.py

```diff
@@ -44,12 +44,20 @@ def mark_calculations(request, user_id=None):
     else:
         xp_per_course = None
 
+    # only show mark ranges where student is enrolled in and is also active
+    user_courses = [sc.course for sc in user.coursestudent_set.all()]
+    assigned_ranges = MarkRange.objects.filter(active=True, courses__in=user_courses)  # filter for mark ranges with courses assigned
+    all_ranges = MarkRange.objects.filter(active=True, courses=None)  # filter for mark ranges with no courses (mark range for all courses)
+
+    markranges = (assigned_ranges | all_ranges).order_by('minimum_mark')
+
```

Comment:
Can you add a bit more commenting to the lines above? It's not clear to me why your doing why your doing.  Your current comments are basically restating the code, which is ok, but more important is why your doing what you're doing.

` (assigned_ranges | all_ranges)` This could use a comment. What's this doing? Why?  I could guess, but not familiar with it...  or?

Reply:
Removed redundant comments
Added comment for (assigned_ranges | all_ranges)